### PR TITLE
feat(policy): versioned ChatGPT UI pool policy with lint and CI gate

### PR DESCRIPTION
**Pull request by Th0rgal**
## Summary
- Adds a versioned (1.0.0) operational policy for the `chatgpt_ui` pool: `docs/policy/CHATGPT_UI_POOL_POLICY.md` (human-readable), `chatgpt_ui_pool_policy.json` (machine-readable), and a JSON Schema.
- Encoded rules: capacity is the **live** `profile_dirs` slot count (never hardcoded); concurrent **read-only** `gpt-5.6-pro` lanes allowed only on disjoint slots with `writer: false`; `compatibility=chatgpt-ui-v2` failures retried **at most once on a different healthy slot**; `auth_required` is **never blind-retried**; `rate_limited` waits for allowance; **never concurrent writers** per workspace; **independent validation before Lean writers**.
- `scripts/policy_lint.py` (stdlib-only) validates the policy against the schema, pins the non-negotiable invariants in code, and cross-checks limits/signals against `src/api/runners/chatgpt_ui.rs`, `scripts/chatgpt_ui_driver.py` (`COMPAT_VERSION`), `docs/CHATGPT_UI_HARNESS.md`, and the skill frontmatter — so policy cannot silently drift from runtime.
- Versions `skills/hermes-mission-control/SKILL.md` (frontmatter `version: 1.0.0`, `metadata.policy_version`) and adds a binding "ChatGPT UI pool policy" section.
- Fixes stale harness doc claim "clamped to 30–7200 seconds" (runtime clamps 30–86400, `src/api/runners/chatgpt_ui.rs:339`); the linter now fails CI if that line goes stale again.
- New `policy-lint` CI job: `python3 scripts/policy_lint.py` + `python3 -m unittest scripts/test_policy_lint.py`.

No runtime, routing, SessionDB, dashboard, or deployment files are touched.

## Test plan
- [x] `python3 scripts/policy_lint.py` → `policy-lint: OK`
- [x] `python3 -m unittest scripts/test_policy_lint.py` → 17 tests OK (repo-clean check + 15 mutation regressions)
- [x] `python3 -m unittest scripts/test_chatgpt_ui_driver.py scripts/test_chatgpt_ui_mock_driver.py scripts/test_chatgpt_ui_smoke.py` → 13 tests OK (unchanged suites)
- [x] `ci.yml` and SKILL.md frontmatter parsed as valid YAML
- [ ] Mark ready-for-review to run full CI (draft PRs are skipped by the workflow `if:` guards)

---

> [!NOTE]
> **Medium Risk**
> Changes mission board auto-retry and profile-slot selection for `chatgpt_ui`, which can alter failure/recovery behavior; policy lint and tests reduce doc drift risk but do not fully exercise production pool edge cases.
> 
> **Overview**
> Introduces a **semver-versioned ChatGPT UI pool policy** (human doc, JSON, JSON Schema) and **`scripts/policy_lint.py`** with mutation tests, wired into a new **`policy-lint` CI job**. The linter keeps policy, harness docs, Rust pool/runtime constants, driver `COMPAT_VERSION`, and **`hermes-mission-control`** frontmatter and binding bullets aligned.
> 
> **Runtime behavior** now follows that policy for `chatgpt_ui` board tasks: **no automatic retry** on auth or rate-limit terminals; **at most one retry** on `compatibility=chatgpt-ui-v2`, with failures tagged **`profile_slot`** and the profile pool **excluding that slot** (and preferring healthy alternatives) on respawn/outbox replay. Pool acquisition also **skips quarantined slots without aborting** when another slot can serve, and a broken lock path no longer fails the whole pool immediately.
> 
> Docs fix the harness **timeout accepted range (30–86400)**. Smaller fixes: **Hermes session delivery test** de-flaked polling assertions, and **sandboxed_node** capacity test holds a permit directly instead of racing a shell loop.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 14cce4c42b40e758fa26c9f0eaffee2bbb116c7b. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by Th0rgal** [comment]
@codex review

**Comment by Th0rgal** [comment]
@codex review

**Comment by Th0rgal** [comment]
@codex review

File: docs/policy/CHATGPT_UI_POOL_POLICY.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+telemetry that a different healthy slot is available before submitting the
+single retry.
+
+## 4. Auth failures: never blind-retry
+
+`auth_required` is terminal until an operator re-provisions login
+interactively in the affected profile. Zero automatic retries — a blind retry
+cannot re-authenticate, wastes a slot, and can trip anti-automation controls.
+The same applies per-slot: an auth-failed slot is unhealthy and must not be
+selected for compatibility retries until re-provisioned.
```
[the remark this answers is not in the document]
Comment (Th0rgal) [reply docs/policy/CHATGPT_UI_POOL_POLICY.md]: Fixed in b501a3f4. The policy now matches the implemented safety boundary exactly: auth is terminal with zero automatic retries, the slot has a machine-checked 1800-second quarantine, cooldown expiry only permits a later explicit recovery attempt and is not treated as proof of reprovisioning, and auth-failed slots remain forbidden for the one compatibility retry.

File: docs/policy/CHATGPT_UI_POOL_POLICY.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+entries (deduplicated, canonicalized). There is no static capacity constant in
+policy or code, and none may be introduced in prose: schedulers and assistants
+must read the live configuration rather than assuming a fixed slot count.
+Each slot is one dedicated browser profile guarded by a cross-process
+exclusive file lock. Acquisition prefers the first available slot with no
+recorded failure, preserving configured order among equivalent candidates.
+A compatibility-failed slot remains usable, but a clean alternative wins the
+next lease. Locked, quarantined, and unavailable slots are never reused; when
+none is healthy and available, the runtime waits until a slot recovers or the
+mission is cancelled.
```
[the remark this answers is not in the document]
Comment (Th0rgal) [reply docs/policy/CHATGPT_UI_POOL_POLICY.md:31]: Fixed in bd61db49 (with warning rate limiting in 8142424a). Lock-path errors are now isolated to that profile, later healthy candidates are tried, and an all-unavailable pool follows the normal wait/cancel path. A regression test covers an unavailable first slot plus a healthy alternative.

File: scripts/policy_lint.py [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+        ("source_of_truth.operator_doc", str(HARNESS_DOC)),
+        ("source_of_truth.policy_doc", str(POLICY_DOC)),
+        ("source_of_truth.skill", str(SKILL_DOC)),
+        ("capacity.source", "profile_dirs"),
+        ("capacity.static_limit", None),
+        ("capacity.acquire_strategy", "first_healthy_available"),
+        ("lanes.read_only_pro.allowed", True),
+        ("lanes.read_only_pro.writer", False),
+        ("lanes.read_only_pro.concurrent", True),
+        ("lanes.read_only_pro.requires_disjoint_slots", True),
```
[the remark this answers is not in the document]
Comment (Th0rgal) [reply scripts/policy_lint.py:94]: Fixed in b501a3f4. `lanes.read_only_pro.model` is now pinned to `gpt-5.6-pro` by both the schema/invariant check, with a mutation test proving that `gpt-4o` is rejected.

File: scripts/policy_lint.py [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+    )
+
+
+def check_versions(repo_root, policy, errors):
+    version = policy.get("version")
+    md_version = doc_version((repo_root / POLICY_DOC).read_text(encoding="utf-8"))
+    if md_version is None:
+        errors.append(f"{POLICY_DOC}: missing 'Version: X.Y.Z' line")
+    elif md_version != version:
+        errors.append(f"{POLICY_DOC}: version {md_version} != policy {version}")
```
[the remark this answers is not in the document]
Comment (Th0rgal) [reply scripts/policy_lint.py:292]: Fixed in b501a3f4. The linter now parses and compares all four Markdown runtime-limit rows (default, range, artifact count, artifact bytes) against the machine policy, with a stale-table mutation test.

**Comment by Th0rgal** [comment]
@codex review

File: scripts/policy_lint.py [hunk 5]
```diff
@@ -0,0 +1,349 @@
...
+        errors.append(
+            f"retry.compatibility_failure.signal {signal!r} != driver "
+            f"compatibility={match.group(1)!r}"
+        )
+
+
+def check_harness_doc(repo_root, policy, errors):
+    doc = (repo_root / HARNESS_DOC).read_text(encoding="utf-8")
+    timeout = policy.get("runtime_limits", {}).get("timeout_secs", {})
+    match = re.search(r"clamped to ([0-9]+)[–-]([0-9]+) seconds", doc)
```
[the remark this answers is not in the document]
Comment (Th0rgal) [reply scripts/policy_lint.py]: Fixed in 68a657fd: the harness now documents the timeout as a validated accepted range and explicitly says out-of-range values are rejected before launch. The policy lint and mutation test now enforce that wording/range.

File: scripts/policy_lint.py [hunk 6]
```diff
@@ -0,0 +1,349 @@
...
+    "integer": int,
+    "null": type(None),
+    "boolean": bool,
+}
+
+
+def validate_schema(instance, schema, path, errors):
+    """Validate the JSON Schema subset used by the policy schema."""
+    if "const" in schema:
+        if instance != schema["const"]:
```
[the remark this answers is not in the document]
Comment (Th0rgal) [reply scripts/policy_lint.py]: Fixed in 68a657fd: schema consts and pinned invariants now use type-strict JSON equality, with regressions proving integer 0 cannot satisfy false and boolean true cannot satisfy integer 1.

File: docs/policy/chatgpt_ui_pool_policy.json [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+  "retry": {
+    "compatibility_failure": {
+      "signal": "compatibility=chatgpt-ui-v2",
+      "max_automatic_retries": 1,
+      "require_different_slot": true,
+      "require_healthy_slot": true
+    },
+    "auth_failure": {
+      "signal": "auth_required",
+      "max_automatic_retries": 0,
```
[the remark this answers is not in the document]
Comment (Th0rgal) [reply docs/policy/chatgpt_ui_pool_policy.json:36]: Fixed in 1665c099. Board settlement now receives the structured terminal reason and suppresses automatic retries for chatgpt_ui AuthError; the task is failed with an operator-reauthentication audit note. Restart reconciliation parses the persisted auth_error reason too. Regression: chatgpt_ui_auth_and_rate_limits_never_auto_retry.

File: docs/policy/chatgpt_ui_pool_policy.json [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+      "operator_action_required": true,
+      "slot_quarantine_secs": 1800
+    },
+    "rate_limited": {
+      "signal": "rate_limited",
+      "max_automatic_retries": 0,
```
[the remark this answers is not in the document]
Comment (Th0rgal) [reply docs/policy/chatgpt_ui_pool_policy.json:42]: Fixed in 1665c099. chatgpt_ui RateLimited now suppresses board auto-retry and settles failed with an allowance-recovery audit note; persisted rate_limited reasons are honored after restart. Covered by the same retry-policy regression.

File: docs/policy/chatgpt_ui_pool_policy.json [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+      "writer": false,
+      "concurrent": true,
+      "requires_disjoint_slots": true
+    }
+  },
+  "retry": {
+    "compatibility_failure": {
+      "signal": "compatibility=chatgpt-ui-v2",
+      "max_automatic_retries": 1,
+      "require_different_slot": true,
```
[the remark this answers is not in the document]
Comment (Th0rgal) [reply docs/policy/chatgpt_ui_pool_policy.json:31]: Fixed in 1665c099. Compatibility errors now carry only the safe slot index; the board persists it in the prior digest and registers a per-mission acquisition requirement before spawn/re-kick. The pool consumes that requirement, excludes the failed slot, and waits for a different healthy slot instead of falling back to the failed or another unhealthy slot. Regression locks the healthy alternative first and proves the retry waits rather than reusing an invalid slot.

File: src/api/runners/chatgpt_ui/profile_pool.rs [hunk 10]
```diff
@@ -93,6 +93,35 @@ fn registry() -> &'static Mutex<HashMap<PathBuf, SlotHealth>> {
...
 
+#[derive(Debug, Clone, Copy)]
+struct ProfileRetryRequirement {
+    excluded_slot: usize,
+    require_healthy: bool,
+}
+
+fn mission_retry_requirements() -> &'static Mutex<HashMap<Uuid, ProfileRetryRequirement>> {
+    static REQUIREMENTS: OnceLock<Mutex<HashMap<Uuid, ProfileRetryRequirement>>> = OnceLock::new();
+    REQUIREMENTS.get_or_init(|| Mutex::new(HashMap::new()))
```
[the remark this answers is not in the document]
Comment (Th0rgal) [reply src/api/runners/chatgpt_ui/profile_pool.rs:104]: Fixed in 14cce4c4. Durable outbox replay now reloads the task prior-result marker and restores the per-mission different-healthy-slot requirement immediately before dispatch, including the coalesced spawn/retry path. The replay regression now uses a chatgpt_ui compatibility retry and asserts slot 4 is restored before the command is queued.

File: scripts/policy_lint.py [hunk 11]
```diff
@@ -0,0 +1,354 @@
...
+        (repo_root / SKILL_DOC).read_text(encoding="utf-8")
+    )
+    if skill_version is None:
+        errors.append(f"{SKILL_DOC}: missing frontmatter 'version:' field")
+    if skill_policy_version is None:
+        errors.append(f"{SKILL_DOC}: missing frontmatter 'policy_version:' field")
+    elif skill_policy_version != version:
+        errors.append(
+            f"{SKILL_DOC}: policy_version {skill_policy_version} != policy {version}"
+        )
```
[the remark this answers is not in the document]
Comment (Th0rgal) [reply scripts/policy_lint.py:304]: Fixed in 14cce4c4. policy_lint now parses each normative skill bullet and derives required model, capacity, retry, cooldown, writer, and validation values from the machine policy. Mutation regressions prove that changing compatibility retry from 1 to 5 or auth retry from 0 to 1 fails lint even when version metadata is untouched.
